### PR TITLE
Firearms skills rework

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

Reworks the firearms skill.

unskilled is now 'SKILL_FIREARMS_CIVILIAN'
default is now 'SKILL_FIREARMS_TRAINED'
trained is now 'SKILL_FIREARMS_EXPERT'

- Civilian skill will allow you to use pistols, SMGs, and certain weapons that have their civilian override variable set to TRUE, such as bolt-action rifles, the ABR-40, the HG-37, or the double barrel shotguns.
- Trained skill is the same as always.
- Same with expert skill. The renames are for readability.

# Explain why it's good for the game

Civilian gun usability is horribly updated. It shouldn't slow your firerate as that makes no sense and makes guns feel terrible to use, and it shouldn't be applied in such a way that it makes pistols and SMGs unusable and the best option running around with a one handed shotgun.

Pistols and SMGs are reasonably newbie-friendly guns for civvies to know how to use, and the civilian shotguns are, well, built for them.

This paves the way for survivors to not be on the same level as marines. If this is an approved idea I can include it here.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
balance: Reworks the firearms skill. Civilians can now fire pistols, SMGs, and certain other civilian weapons without penalties. Civilian gun penalties have had their firerate reduction remove and scatter increased.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun_helpers.dm [hunk 1]
```diff
@@ -938,3 +938,19 @@ As sniper rifles have both and weapon mods can change them as well. ..() deals w
 	if(slot != WEAR_L_HAND && slot != WEAR_R_HAND)
 		return FALSE
 	return TRUE
+
+/**
+ * Returns one of the two override values if either are null, preferring the argument value.
+ * Otherwise, returns TRUE if it is in a civilian usable category (Handguns or SMGs), FALSE if it is not.
+ */
+/obj/item/weapon/gun/proc/is_civilian_usable(mob/user, arg_override)
```
Comment (harryob) [inline code/modules/projectiles/gun_helpers.dm:946]: what's the point of the arg_override arg btw it doesn't seem used

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun_helpers.dm (hunk 1 above)
Comment (carlarctg) [reply code/modules/projectiles/gun_helpers.dm:946]: it's for the future

File: code/__DEFINES/misc.dm [hunk 2]
```diff
@@ -223,6 +223,9 @@
 #define GUN_CATEGORY_SHOTGUN 4
 #define GUN_CATEGORY_HEAVY 5
 
+// These guns can be used at maximum efficacy by untrained civilians.
+#define UNTRAINED_USABLE_CATEGORIES list(GUN_CATEGORY_HANDGUN, GUN_CATEGORY_SMG)
```
Comment (morrowwolf) [inline code/__DEFINES/misc.dm:227]: Not SMGs

**Review by morrowwolf — COMMENTED** [review]
Other than that one thing the concept is good. Remind me to architect approve after the change.

File: code/__DEFINES/misc.dm (hunk 2 above)
Comment (morrowwolf) [reply code/__DEFINES/misc.dm:227]: Spoke on discord, we'll see how it goes.

**Review by harryob — APPROVED** [review]

**Pull request merged**